oauth2client/contrib/appengine.py

```diff
@@ -477,15 +469,15 @@ def _parse_state_value(state, user):
         user: google.appengine.api.users.User, The current user.
 
     Raises:
-        InvalidXsrfTokenError: if the XSRF token is invalid.
+        ValueError: if the XSRF token is invalid.
```

Comment:
This is done behind the scenes during the OAuth callback.  If the client and server tokens differ, the user should would just see the "request failed" response, and that 's all that's being checked here.

I'm of the opinion of keeping the InvalidXsrfTokenError for clarity, but @jonparrott said that since that exception is only used internally, to not have a separate custom exception, which also makes sense.

In this case, it is not the result of a programming defect or user error, but rather, a potential CSRF/XSRF vulnerability.